aten/src/ATen/native/UpSampleBicubic2d.cpp

```diff
@@ -24,7 +24,7 @@ static void upsample_bicubic2d_out_frame(
         const scalar_t* in = &idata[output_y * input_width + output_x];
         scalar_t* out = &odata[output_y * output_width + output_x];
 
-        for (int64_t c = 0; c < channels; ++c) {
+        for (int64_t c = 0; c < channels * nbatch; ++c) {
```

Comment:
@Enealor Do we need to modify the next line as well? (I'm pinging you because you sound like you know something about this kernel: https://github.com/pytorch/pytorch/issues/32918#issuecomment-581755058). The next line looks fishy to me but I don't know what is going on this kernel too well.

Reply:
I don't believe the line needs to be modified. The kernel is supposed to copy information only in this case, and that is what is happening as far as I can tell. To be sure I'm understanding the logic right:
1. For each [x, y], we grab the address of the input and output scalar at that [x, y].
1. We copy that over with `out[0] = in[0]`.
1. We use pointer arithmetic to move to the address of the input and output data in the next channel, and repeat 2. (I think this might cause an issue for non-contiguous data, but I haven't tested this.)

Additionally, these changes produced the correct results with regards to the issue you linked. Did I address your concerns?

Reply:
Oh I see. Yes, you're right, thank you for the explanation.

Yeah, it looks like the code assumes the data is contiguous, but maybe there is a .contiguous() call somewhere upstream. At any rate I'll write up a test case for this and then it should be good to go.

Reply:
Following line makes the input contiguous.

https://github.com/pytorch/pytorch/blob/0ea1abe07bfa4377fa92d2f9222a0a0eed095b68/aten/src/ATen/native/UpSampleBicubic2d.cpp#L198-L201